src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1863,18 +1862,27 @@ def model(self, model: torch.nn.Module) -> None:
     """
 
     @property
-    def log_dir(self) -> Optional[str]:
+    def experiment_dir(self) -> str:
         if len(self.loggers) > 0:
-            if isinstance(self.loggers[0], TensorBoardLogger):
-                dirpath = self.loggers[0].log_dir
+            if self.loggers[0].save_dir is not None:
+                save_dir = self.loggers[0].save_dir
             else:
-                dirpath = self.loggers[0].save_dir
+                save_dir = self.default_root_dir
+            dirpath = os.path.join(save_dir, self.loggers[0].experiment_dir)
         else:
             dirpath = self.default_root_dir
 
         dirpath = self.strategy.broadcast(dirpath)
         return dirpath
 
+    @property
+    def log_dir(self) -> str:
```

Comment:
Where did we discuss that `log_dir` would be deprecated?

Reply:
If they have different purposes, then I don't think `log_dir` should be deprecated. There would be uses cases for each.

If they have the same purpose and just differ in implementation, then we could just change it without a deprecation and keep the `log_dir` name.
In the past, we've done changes to `log_dir` without going through a deprecation.

Reply:
@carmocca The reasoning is the following:

I think we should make `log_dir` do what it does for our default logger, TensorBoard, meaning returning the full log dir `root / name / version`. This would mean calling log_dir after this PR is the same for these users, no breaking change.

BUT, the consideration for deprecation is only relevant for everyone else who doesn't use TB. Here, we have the option to 
- breaking change, no warning, just change it
- forward deprecation: warn that the behavior will change in the future. Problem: User can't do anything to avoid the warning
- Deprecate the entire log_dir: Same as above, but user can change to trainer.experiment_dir to avoid the deprecation warning

Reply:
There is also the cheap alternative to masquerade it as a fix, i.e., pushing the change as "Fixed: Trainer.log_dir is now consistently returning the versioned log dir for all loggers".

Reply:
Thanks for the context. I think you should choose.

My only suggestion is that this is hard to understand. We would have `log_dir`, `experiment_dir`, `save_dir`, `default_root_dir` ... (any other dir?) and the relationship is not clear from just the names.
If the solution can avoid introducing a new name, I would vote for it. `log_dir` is for the `Trainer`, but would it make sense to have it also be the name for `Logger.log_dir`?

Reply:
Hi, any update on it?
I vote for deprecate the entire `log_dir` or  keep `log_dir`

Reply:
I lean towards the latter. @awaelchli what's your opinion? cc @justusschock too

Reply:
@carmocca It is not just about picking the name. The main challenge is how to deliver the improved way without major breaking changes. We can imagine that logging and managing experiments is a core Lightning experience, so it will affect many users. Introducing experiment dir was meant to minimize that (existing code still works). I would also love to keep the same name log_dir, but how are we going to do it???

Reply:
I would also try to keep the naming as unified as possible and go with `Logger.log_dir`.